packages/manager/apps/pci-load-balancer/src/api/hook/useL7Policy.tsx

```diff
@@ -118,3 +122,36 @@ export const useDeletePolicy = ({
     ...mutation,
   };
 };
+
+type CreatePolicyProps = {
+  projectId: string;
+  listenerId: string;
+  region: string;
+  policy: TL7Policy;
+  onError: (cause: Error) => void;
+  onSuccess: (newPolicy: TL7Policy) => void;
+};
+
+export const useCreatePolicy = ({
+  projectId,
+  listenerId,
+  policy,
+  region,
+  onError,
+  onSuccess,
+}: CreatePolicyProps) => {
+  const mutation = useMutation<TL7Policy>({
+    mutationFn: async () => createPolicy(projectId, region, listenerId, policy),
+    onError,
+    onSuccess: async (newPolicy) => {
+      await queryClient.invalidateQueries({
```

Comment:
May be put onSuccess first, invalidateQuery is asynchronous, i experienced cases where it take some time to execute witch delays the navigate that comes afterwards, and i had bugs tickets because of that, so either call it after the onSuccess or simply remove the await